Test your web component in a web page

fgravin opened the pull request:
- Add a npm task `serve-wc`
```
npm run serve-wc -- webcomponent_name) [--build]
```
to build a Web Component and immediatly serve it in a web page, on  http://127.0.0.1:8001

This task shows all the steps needed to export and use your web component in a standalone web site.

- Fix `storybook-wc` task to export and test your Web Components in storybook.

jahow commented on CONTRIBUTING.md:
````diff
@@ -74,37 +75,34 @@ export class AppModule {
-    <gn-search-snapshot></gn-search-snapshot>
-```
-and run your html file in a web server
+#### Test your Web Component
````
I think the documentation is much better now, but it would be better suited in `webcomponents/README` instead of `CONTRIBUTING`. Especially the general usage part.

jahow commented on tools/build/webcomponent/serveWebComponent.sh:
```diff
@@ -0,0 +1,21 @@
+DIST_PATH=webcomponents/${APP_NAME}/dist/
+DIST_WC_PATH=$DIST_PATH'webcomponents/'
+
+if [ ${3} ] && [ $3} = "--build" ]
```
```suggestion
if [ ${3} ] && [ ${3} = "--build" ]
```

jahow commented on CONTRIBUTING.md:
````diff
@@ -74,37 +75,34 @@ export class AppModule {
+To test your web component in true a web page, run
 ```shell script
- "serve-wc": "http-server --gzip",
+npm run serve-wc -- (application_name) (webcomponent_tag_name) [--build]
````
I think the logic here is a bit misleading... I had trouble understanding what was the difference between `application_name` and `webcomponent_tag_name`. It's working for now but maybe we should have a list of webcomponents somewhere with their respective angular app and tag name, and read from that instead?

fgravin replied:
I have refactored the whole thing to make web component applications simpler, also angular application name and WC tag name should be the same

jahow commented on webcomponents/search-wc/.browserslistrc:
```diff
@@ -5,7 +5,7 @@
 #   npx browserslist
 
-> 0.5%
+> 5%
```
what was the motivation for this btw? web component support?

fgravin replied:
It's to avoid Angular to generate to many output files (ie maines5.js, maines2015.js etc..).

jahow reviewed (approved):
Just a typo to fix in the bash script, and some suggestions re: documentation. Other than that, the setup is now much better for working on web components :)

fgravin commented:
Thanks @jahow for the review, I have refactored the web component application to be simpler to create and export.
I have added a `npm run build:wc` to build all web components, a prerequisite to run `storybook-wc`.
Also updated and moved the web component documentation.

fgravin commented:
@jahow this is ready to merge
